src/main/java/com/pinterest/secor/common/SecorConfig.java

```diff
@@ -277,6 +277,10 @@ public long getMaxFileAgeSeconds() {
         return getLong("secor.max.file.age.seconds");
     }
 
+    public int getMaxActiveFiles() {
+        return getInt("secor.max.file.count", 100);
```

Comment:
How about we set the default to -1, and on the caller side if getMaxActiveFiles returns -1 it will not forceUpload, this way we can maintain the backward compatible behavior for the installations who didn't face memory problem.